db/src/main/kotlin/edu/artic/db/AppDataManager.kt

```diff
@@ -1,38 +1,221 @@
+                    is ProgressDataState.Done<*> -> {
+                        loadPostAppData()
+                                .subscribe({ amountDownload ->
+                                    Log.d("AppDataManager", "Amount downloaded $amountDownload")
+                                    ProgressDataState.Downloading(.34f + amountDownload * .33f)
+                                    if (amountDownload == 2) {
+                                        observer.onNext(ProgressDataState.Done(true))
+                                    }
+                                }, {
+                                    observer.onError(it)
+                                }, {
+                                    observer.onComplete()
+                                })
+                    }
+                    is ProgressDataState.Empty -> {
+                        loadPostAppData()
+                                .subscribe({ amountDownload ->
+                                    Log.d("AppDataManager", "Amount downloaded $amountDownload")
+                                    ProgressDataState.Downloading(.34f + amountDownload * .33f)
+                                    if (amountDownload == 2) {
+                                        observer.onNext(ProgressDataState.Done(true))
+                                    }
+                                }, {
+                                    observer.onError(it)
+                                }, {
+                                    observer.onComplete()
+                                })
+                    }
+                    is ProgressDataState.Downloading -> {
+                        observer.onNext(ProgressDataState.Downloading(it.progress * .33f))
+                    }
+                }
+
+            }, {
+                observer.onError(it)
+            }, {
+
+            }
+            )
+        }
+    }
+
+    fun getBlob(): Observable<ProgressDataState> {
         return serviceProvider.getBlobHeaders()
+                .flatMap { headers ->
+                    if (!headers.containsKey(HEADER_LAST_MODIFIED) || headers[HEADER_LAST_MODIFIED]?.get(0)
+                            != appDataPreferencesManager.lastModified) {
+                        serviceProvider.getBlob()
+                    } else {
+                        ProgressDataState.Empty.asObservable()
+                    }
+                }.flatMap { appDataState ->
+                    if (appDataState is ProgressDataState.Done<*>) {
+                        //Save last downloaded headers
+                        appDataState.headers[HEADER_LAST_MODIFIED]?.let {
+                            appDataPreferencesManager.lastModified = it[0]
+                        }
+
+                        val result = appDataState.result as ArticAppData
+
+                        try {
+                            appDatabase.dashboardDao.setDashBoard(result.dashboard)
+                        } catch (e: Throwable) {
```

Comment:
try catching `throwable` suppresses any  exception thrown here. Should we catch specific exceptions here? or throw exception on Debug builds so we can catch errors here fast and early. I can foresee us seeing blank data and not realizing it until much later and spending a lot of time figuring out why something is going wrong.

Reply:
So the try catch stuff was not passing along data to onError's before I can remove it for this build.